packages/lesswrong/components/posts/TableOfContents/TableOfContentsList.tsx

```diff
@@ -54,8 +56,14 @@ const TableOfContentsList = ({tocSections, title, onClickSection, displayOptions
         search: isEmpty(query) ? '' : `?${qs.stringify(query)}`,
         hash: `#${anchor}`,
       });
-      let sectionYdocumentSpace = anchorY + window.scrollY;
-      jumpToY(sectionYdocumentSpace);
+
+      // This is forum-gating of a fairly subtle change in scroll behaviour, LW may want to adopt scrollFocusOnElement
+      if (!isLWorAF) {
+        scrollFocusOnElement({ id: anchor, options: {behavior: "smooth"}})
```

Comment:
I wasn't able to reproduce this exactly. I was able to reproduce the mutation observer firing lots of times, but not the stuttering. But to me it looks like the smooth scroll animation is restarting on every new call to scrollTo (whereas browsers I'm testing this one don't do this, maybe they are on newer versions?).

Anyway I've attempted to work around this in [this commit](https://github.com/ForumMagnum/ForumMagnum/pull/9699/files/0091d1376ebc9840819c4ed074be030f66cbef26..cc4379798535e724fa5531f5641f9c8d3658d4ab), by:
- Only re-firing `scrollTo` if the target position has changed by more than 40px since the previous time (so if there isn't much layout shift it should only fire once)
- Using instant scrolling if it does need to re-fire

Reply:
Nice, I tested it and it looks smooth for me now :)